packages/flutter/lib/src/services/keyboard_key.dart

```diff
@@ -238,6 +238,21 @@ class LogicalKeyboardKey extends KeyboardKey {
     return result == null ? <LogicalKeyboardKey>{} : <LogicalKeyboardKey>{result};
   }
 
+  /// Takes a set of keys, and returns the same set, but with any keys that have
+  /// synonyms replaced.
+  ///
+  /// It is used, for example, to make sets of keys with members like
+  /// [controlRight] and [controlLeft] and convert that set to contain just
+  /// [control], so that the question "is any control key down?" can be asked.
+  static Set<LogicalKeyboardKey> collapseSynonyms(Set<LogicalKeyboardKey> input) {
```

Comment:
I love static methods.  This could even be directly unit tested if you want.

Reply:
LOL!  Yeah, ok, added a test.  Very subtle. :-)

Reply:
😄